### PR TITLE
Use the AggTestConfig object in testCase

### PR DESCRIPTION
**Pull request by not-napoleon**
This is a second attempt at https://github.com/elastic/elasticsearch/pull/90320.  I've left out the more challenging parts of that PR, notably reconciling the buildIndex callback with aggregations that want to pass in searchers directly and making the `aggTestConfig` object generic over the verify method.  For the verify method, I just didn't touch it.  For the buildIndex/Searcher conflict, I've taken a different approach: I factored the searcher out of the config object, and pass it directly to search and reduce.  Likewise test case can just accept the builder and use it to make a searcher, which it passes to search and reduce. 

So what's the point of this, you may ask.  The point is that now both `testCase` and `searchAndReduce` use the same config object for many of their common settings.  This will make life easier converting bespoke tests to use one or the other, and thus get the memory checking benefits of routing through `searchAndReduce` without requiring endless configuration overrides.  This change should remove the need for [this config function](https://github.com/elastic/elasticsearch/pull/90447/files#diff-330f7d4890840fa1508bd7942db7c39e84641b7a1a260be53c58042b70902413R437) for example (/cc @romseygeek ).

Once again, I apologize for the huge diff from a largely automated refactor.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/prefix/IpPrefixAggregatorTests.java [hunk 1]
```diff
@@ -1044,7 +1054,7 @@ public void testAggregationWithQueryFilter() throws IOException {
                 ipPrefix.getBuckets().stream().sorted(IP_ADDRESS_KEY_COMPARATOR).map(InternalIpPrefix.Bucket::getDocCount).toList(),
                 List.of(4L)
             );
-        }, fieldType);
+        }, new AggTestConfig(aggregationBuilder, fieldType));
```
Comment (martijnvg) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/prefix/IpPrefixAggregatorTests.java:1057]: I think `withQuery(query)` needs to be added here?

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java [hunk 2]
```diff
@@ -470,7 +466,7 @@ public void testNestedOrdering() throws IOException {
                 nestedBuilder.subAggregation(maxAgg);
                 termsBuilder.subAggregation(nestedBuilder);
 
-                terms = searchAndReduce(new AggTestConfig(newSearcher(indexReader, false, true), termsBuilder, fieldType1, fieldType2));
+                terms = searchAndReduce(newSearcher(indexReader, false, true), aggTestConfig);
```
Comment (martijnvg) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java:469]: I think `aggTestConfig = new AggTestConfig(termsBuilder, fieldType1, fieldType2);` needs to be added before this line? `termsBuilder` is changed in the lines above here, so a new `AggTestConfig ` instance should be created?

**Review by martijnvg — APPROVED** [review]
LGTM 👍

**Comment by not-napoleon** [comment]
@elasticmachine update branch

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/prefix/IpPrefixAggregatorTests.java (hunk 1 above)
Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/prefix/IpPrefixAggregatorTests.java:1057]: yup, looks like I dropped that by accident.

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java (hunk 2 above)
Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java:469]: yup, well spotted.

**Pull request merged**